### PR TITLE
add nonreentrant to supplyTokenTo and redeemToken function

### PR DESCRIPTION
**Pull request by kamescg**
Issue: https://github.com/code-423n4/2021-06-pooltogether-findings/issues/119

### CODE REVIEW

**Review by PierrickGT — APPROVED** [review]

File: contracts/SushiYieldSource.sol [hunk 1]
```diff
@@ -4,13 +4,14 @@ pragma solidity 0.6.12;
...
 import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
 
 import "./ISushiBar.sol";
 import "./ISushi.sol";
 
 /// @title A pooltogether yield source for sushi token
 /// @author Steffel Fenix
-contract SushiYieldSource is IYieldSource {
+contract SushiYieldSource is IYieldSource, ReentrancyGuard {
```
Comment (asselstine) [inline contracts/SushiYieldSource.sol:14]: You must initialize the guard

**Review by asselstine — CHANGES_REQUESTED** [review]

File: contracts/SushiYieldSource.sol (hunk 1 above)
Comment (PierrickGT) [reply contracts/SushiYieldSource.sol:14]: There is no need to initialize it when using the non upgradeable version one no?

Comment (PierrickGT) [reply contracts/SushiYieldSource.sol:14]: @asselstine there is no need to initialize it when using the non upgradeable version one no?

Comment (asselstine) [reply contracts/SushiYieldSource.sol:14]: The constructor must be called

Comment (PierrickGT) [reply contracts/SushiYieldSource.sol:14]: Done: https://github.com/pooltogether/sushi-pooltogether/pull/10/commits/d8be0a80ea8af97938fe07b83f272c7cce166ec2

**Review by asselstine — APPROVED** [review]

**Pull request merged**